src/audio/CMakeLists.txt

```diff
@@ -118,16 +121,17 @@ check_optimization(fma -mfma -DOPS_FMA)
 check_optimization(hifi2ep -mhifi2ep -DOPS_HIFI2EP)
 check_optimization(hifi3 -mhifi3 -DOPS_HIFI3)
 
-set(sof_audio_modules volume src asrc eq-fir eq-iir dcblock crossover)
+set(sof_audio_modules volume src asrc eq-fir eq-iir dcblock crossover tdfb)
 
 # sources for each module
 set(volume_sources volume/volume.c volume/volume_generic.c)
 set(src_sources src/src.c src/src_generic.c)
 set(asrc_sources asrc/asrc.c asrc/asrc_farrow.c asrc/asrc_farrow_generic.c)
-set(eq-fir_sources eq_fir/eq_fir.c eq_fir/fir.c)
+set(eq-fir_sources eq_fir/eq_fir.c eq_fir/eq_fir_generic.c)
 set(eq-iir_sources eq_iir/eq_iir.c eq_iir/iir.c)
 set(dcblock_sources dcblock/dcblock.c dcblock/dcblock_generic.c)
 set(crossover_sources crossover/crossover.c crossover/crossover_generic.c)
+set(tdfb_sources tdfb/tdfb.c tdfb/tdfb_generic.c)
```

Comment:
how are you linking this against fir?

Reply:
I don't know, I'm not good with cmake... I assume the maths library where FIR is by default is included to link of SOF. I haven't seen issues with SOF images and testbench build. But there was one case where link with cmake did the build and actual compile phase was skipped so I need to check this is not a similar case (by observing build shell text output). But is such check enough?

Reply:
@singalsu pls work with @juimonen here, there should be a Kconfig dependency/select already for FIR users so that if any FIR client is enabled then this would also build FIR core. It's also possible in your testing that FIR is already being built by another client.

Reply:
Yep, I noticed myself the mistake now. It's easy to fix and test, at least the make menuconfig ui level issue.

Reply:
The fix is here, sorry for missing you point. This makes the make menuconfig UI work logically. MATH_FIR is enabled when FIR EQ or TDFB is enabled.

```
diff --git a/src/audio/Kconfig b/src/audio/Kconfig
index 56dd2d858330..f3b04e893afb 100644
--- a/src/audio/Kconfig
+++ b/src/audio/Kconfig
@@ -302,7 +302,7 @@ endif # COMP_ASRC
 
 config COMP_TDFB
        bool "TDFB component"
-        depends on COMP_FIR
+       select MATH_FIR
        default y
        help
          Select for time domain fixed beamformer (TDFB) component. The

```
CMakeLists.txt in math directory already contained this:

```
if(CONFIG_MATH_FIR)
        add_local_sources(sof fir_generic.c fir_hifi2ep.c fir_hifi3.c)
endif()

```

Reply:
Done, this fix is in current version.